src/Core/Types.Filters/Array/ArrayFilterFieldDescriptor.cs

```diff
@@ -0,0 +1,152 @@
+
+        public ArrayFilterFieldDescriptor(
+            IDescriptorContext context,
+            PropertyInfo property,
+            Type type)
+            : base(context, property)
+        {
+            _type = type;
+            AllowedOperations = new HashSet<FilterOperationKind>
+            {
+                FilterOperationKind.ArraySome,
+                FilterOperationKind.ArrayNone,
+                FilterOperationKind.ArrayAll,
+                FilterOperationKind.ArrayAny
+            };
+        }
+
+        protected override ISet<FilterOperationKind> AllowedOperations { get; }
+
+        /// <inheritdoc/>
+        public new IArrayFilterFieldDescriptor BindFilters(
+            BindingBehavior bindingBehavior)
+        {
+            base.BindFilters(bindingBehavior);
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public IArrayFilterFieldDescriptor BindExplicitly() =>
+            BindFilters(BindingBehavior.Explicit);
+
+        /// <inheritdoc/>
+        public IArrayFilterFieldDescriptor BindImplicitly() =>
+            BindFilters(BindingBehavior.Implicit);
+
+
+        protected override FilterOperationDefintion CreateOperationDefinition(
+            FilterOperationKind operationKind)
+        {
+            if (FilterOperationKind.ArrayAny == operationKind)
+            {
+                return CreateBooleanOperation(operationKind).CreateDefinition();
+            }
+            else
+            {
+                return CreateOperation(operationKind).CreateDefinition();
+            }
+
+        }
+
+
+        protected FilterOperation GetFilterOperation(
+            FilterOperationKind operationKind)
+        {
+            return new FilterOperation(
+                _type,
+                operationKind,
+                Definition.Property);
+        }
+
+        protected ClrTypeReference GetTypeReference()
+        {
+            return new ClrTypeReference(
+                    typeof(FilterInputType<>).MakeGenericType(_type),
+                        Definition.Type.Context,
+                        true,
+                        true
+           );
+        }
+
+
+
+        private ArrayFilterOperationDescriptor CreateOperation(
+            FilterOperationKind operationKind)
+        {
+            var operation = GetFilterOperation(operationKind);
+            var typeReference = GetTypeReference();
+            return ArrayFilterOperationDescriptor.New(
+                Context,
+                this,
+                CreateFieldName(operationKind),
+                typeReference,
+                operation);
+            ;
+        }
+
+        private ArrayBooleanFilterOperationDescriptor CreateBooleanOperation(
+          FilterOperationKind operationKind)
+        {
+            var operation = GetFilterOperation(operationKind);
+
+            var typeReference = RewriteTypeToNullableType(
+                new ClrTypeReference(typeof(bool),
```

Comment:
@michaelstaib If you know a better way of getting the ClrType Reference here, let me know :D